server_routers.go

```diff
@@ -412,3 +414,83 @@ func (d clientRouter) FindProviders(ctx context.Context, cid cid.Cid, limit int)
 func (d clientRouter) FindPeers(ctx context.Context, pid peer.ID, limit int) (iter.ResultIter[*types.PeerRecord], error) {
 	return d.Client.FindPeers(ctx, pid)
 }
+
+var _ server.ContentRouter = sanitizeRouter{}
+
+type sanitizeRouter struct {
+	router
+}
```

Comment:
The other option is to apply `filterPrivateMultiaddr` directly in `libp2pRouter`, and the `iter.Maps` in `clientRouter`.